core/trino-main/src/main/java/io/trino/server/CoordinatorModule.java

```diff
@@ -363,7 +363,11 @@ List<OutputStatsEstimatorFactory> getCompositeOutputDataSizeEstimatorDelegateFac
         install(new QueryExecutionFactoryModule());
 
         // cleanup
-        binder.bind(ExecutorCleanup.class).asEagerSingleton();
+        executorServiceCleanupBinder(binder)
```

Comment:
There are more places that need this change. Are you going to update them as well?

Reply:
I see https://github.com/trinodb/trino/pull/20960#issuecomment-1981172170. You can resolve this comment.

Reply:
Fixed.